lib/validate.js

```diff
@@ -431,7 +431,147 @@ function validateArchiveWriteStream(req, res, next) {
     next();
 }
 
+/*
+ * Validates that the query string parameter "paramName" is a valid signal,
+ * and calls "callback" with an error parameter and the validated value. For
+ * consistency, the validated value is always a string, even if the signal is
+ * represented by a number.
+ */
+function validateSignal(req, paramName, callback) {
+    assert.object(req, 'req');
+    assert.object(req.query, 'req.query');
+    assert.string(paramName, 'paramName');
+    assert.func(callback, 'callback');
+
+    var signalString;
+    var validSignal = false;
+    var err;
+
+    // XXX This includes all signals from process.binding('constants') plus
+    // the special '0' signal. Anything not on this list is not supported by
+    // the backend, but it's possible we'll want to remove some from this list
+    // too as some of these probably don't make sense to send.
+    var validSignals = [
+        'SIGABRT', 'SIGALRM', 'SIGBUS', 'SIGCHLD', 'SIGCONT', 'SIGFPE',
+        'SIGHUP', 'SIGILL', 'SIGINT', 'SIGIO', 'SIGIOT', 'SIGKILL', 'SIGLOST',
+        'SIGPIPE', 'SIGPOLL', 'SIGPROF', 'SIGPWR', 'SIGQUIT', 'SIGSEGV',
+        'SIGSTOP', 'SIGSYS', 'SIGTERM', 'SIGTRAP', 'SIGTSTP', 'SIGTTIN',
+        'SIGTTOU', 'SIGURG', 'SIGUSR1', 'SIGUSR2', 'SIGVTALRM', 'SIGWINCH',
+        'SIGXCPU', 'SIGXFSZ', 'ABRT', 'ALRM', 'BUS', 'CHLD', 'CONT', 'FPE',
+        'HUP', 'ILL', 'INT', 'IO', 'IOT', 'KILL', 'LOST', 'PIPE', 'POLL',
+        'PROF', 'PWR', 'QUIT', 'SEGV', 'STOP', 'SYS', 'TERM', 'TRAP', 'TSTP',
+        'TTIN', 'TTOU', 'URG', 'USR1', 'USR2', 'VTALRM', 'WINCH', 'XCPU',
+        'XFSZ', '0', '1', '2', '3', '4', '5', '6', '6', '8', '9', '10', '11',
```

Comment:
@trentm pointed out in sdc-vmapi#22 that there are duplicates. Looks like the same problem exists here.